tools/crm_error.c

```diff
@@ -120,17 +120,33 @@ main(int argc, char **argv)
     }
 
     if (options.do_list) {
-        int start, end, width;
+        int start, end;
+
+        /* Get length of longest (most negative) standard Pacemaker return code
+         * This should be longer than all the values of any other type of return
+         * code.
+         */
+        char *longest_rc_str = pcmk__itoa(pcmk_rc_error - pcmk__n_rc + 1);
+        int code_width = strlen(longest_rc_str);
+        int name_width = 0;
```

Comment:
Oh that is neat.

Reply:
> Should use size_t for all these

They have to be `int`s for the width specifier. The return values are `size_t` but we're not using them for anything other than width specifiers. So I think it makes more sense to use an `int` variable than to use a `size_t` and cast back to `int` in the `printf()` call.

```
pcmk_output.c:1963:34: error: field width specifier ‘*’ expects argument of type ‘int’, but argument 3 has type ‘size_t’ {aka ‘long unsigned int’} [-Werror=format=]
 1963 |         return out->info(out, "% *d: %-*s  %s", code_width, code, name_width, name,
      |                                ~~^~             ~~~~~~~~~~
      |                                  |              |
      |                                  int            size_t {aka long unsigned int}
```

Reply:
Ah right. Static analysis is the main reason to care, since we know it's not a problem here, so I don't mind leaving them as ints. But in case you're interested there are two portable approaches:

* use size_t with %lld and cast it to long long.
* use size_t for initial values, error if they're out of int range, and cast them to int otherwise.